MockerTests/MockerTests.swift

```diff
@@ -401,4 +408,126 @@ final class MockerTests: XCTestCase {
         
         waitForExpectations(timeout: 10.0, handler: nil)
     }
+
+	/// It should call the `onMockFor` filter closure if set.
+	func testOnMockForFilterCalled() {
+        let url = URL(string: "https://www.fakeurl.com/DynamicTests")!
+		var onMockForCalled = false
+		
+		Mocker.shared.onMockFor = { (request: URLRequest) -> Mock? in
+			onMockForCalled = true
+			return nil
+		}
+		
+        var mock = Mock(url: url, dataType: .json, statusCode: 200, data: [.get: Data()])
+        let expectation = expectationForCompletingMock(&mock)
+        mock.register()
+
+		URLSession.shared.dataTask(with: URLRequest(url: url)).resume()
+		
+        wait(for: [expectation], timeout: 2.0)
+
+		XCTAssert(onMockForCalled, "our onMockFor handler was not called!")
+	}
```

Comment:
Mix of TAB and spaces for indentation makes this piece of code look strangely indented,